reports/_model.qmd

```diff
@@ -12,6 +12,282 @@ training_data_pred <- training_data %>%
+}
+
+make_township_ratio_decile_graph <- function(data, township){
+  
+# Manually set axes to highest and lowest contained in the data
+y_axis_values <- data %>%
+  filter(geography_type == 'township_code',
+           by_class == FALSE,
+           triad_code == run_triad_code,
+           num_quantile == '10') %>% 
+  summarise(min_value = min(median_ratio),
+             max_value = max(median_ratio))
+  
+  
+  data_to_plot <- data %>%
+    filter(
+      geography_type == 'township_code',
+      by_class == FALSE,
+      triad_code == run_triad_code,
+      num_quantile == '10',
+      township_name == township
+    ) %>%
+    select(township_name, quantile, median_ratio, lower_bound, upper_bound) %>%
+    mutate(lower_bound_short = sapply(lower_bound, shorten_number),
+         upper_bound_short = sapply(upper_bound, shorten_number),
+         custom_label = paste(quantile, 
+                            "\n[$", 
+                            lower_bound_short,
+                            "K",
+                            "-\n",
+                            "$",
+                            upper_bound_short,
+                            "K",
+                            "]",
+                            sep=""))
+  
+  print(
+    ggplot(data_to_plot, aes(x=quantile, y=median_ratio)) +
+    geom_rect(aes(xmin=-Inf, xmax=Inf, ymin=0.95, ymax=1.05),
+              fill="#D3F2C2", alpha=0.1) +
+    geom_line(aes(group=1), color="#2C3E50") +
+    geom_point(color="deepskyblue4", size=3) +
+    geom_text(aes(label=round(median_ratio, 2)),
+              vjust = -0.55,
+              color = "#34495E",
+              size = 3.9) +
+    geom_hline(yintercept=1, linetype="dashed", color="darkgray") +
+    theme_minimal() +
+    labs(title=township,
+         x="Decile", y="Median Ratio") +
+      # 0.08 is appended to the y_axis value so that the
+      # highest value label for the point isn't cut off
+      coord_cartesian(ylim = c(y_axis_values$min_value,
+                               y_axis_values$max_value + 0.08)) +
+      scale_x_continuous(breaks = 1:10, labels = data_to_plot$custom_label) +
+      theme(axis.text.x = element_text(size = 10))  
+  )
+}
+
+```
+
+```{r}
+# Get township name for graphs
+data_for_tabset <- model_performance_quantile_test %>%
+  left_join(ccao::town_dict, by = c("geography_id" = "township_code")) %>% 
+    select(-triad_code.y) %>%
+  rename(triad_code = triad_code.x)
+
+# Collect townships to iterate over
+townships_to_iterate <- data_for_tabset %>%
+  filter(geography_type == 'township_code',
+         by_class == FALSE,
+         triad_code == run_triad_code,
+         num_quantile == '10') %>%
+  select(township_name) %>%
+  unique() %>%
+  pull(township_name)
+```
+
+::: {.panel-tabset}
+
+```{r, results='asis'}
+# Dynamically produce tabset
+for (township in townships_to_iterate) {
+  cat("##", township, "\n")
+  make_township_ratio_decile_graph(data_for_tabset, township)
+  cat("\n\n")
```

Comment:
Dynamically creates township tabset based on whichever triad the model is working with.

Reply:
Nice! @Damonamajor you can use this same setup to close #94.